make sure that if value is string but metadata is broadleaf enumeration final value will be a friendly type of broadleaf enumeration

oleksiimiroshnyk opened the pull request:
- make sure that if value is string but metadata is broadleaf enumeration final value will be a friendly type of broadleaf enumeration

Fixes: BroadleafCommerce/QA#4846

Commit:
BroadleafCommerce/QA#4846
- make sure that if value is string but metadata is broadleaf enumeration final value will be a friendly type of broadleaf enumeration

KatrukOV reviewed (approved):
We should improve this (add a method to the interface and not use reflection) in 6.3

Commit:
BroadleafCommerce/QA#4846
- Make sure that toString will not be invoked on enumeration class instance, so no abc@123 strings will be present
